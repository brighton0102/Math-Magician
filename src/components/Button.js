import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

const Button = ({ label, type, onClick }) => (
  <button className={`btn-item ${type}`} type="button" onClick={onClick}>
    {label}
  </button>
);

Button.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired, // 'type' is optional with a default value
  onClick: PropTypes.func.isRequired,
};

export default Button;
