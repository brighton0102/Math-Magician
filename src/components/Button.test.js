import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Button from './Button';

test('Button snapshot test', () => {
  const component = renderer.create(
    <Button label="Test Button" type="custom" onClick={() => {}} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
