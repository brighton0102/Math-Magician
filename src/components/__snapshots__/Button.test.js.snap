// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot test 1`] = `
<button
  className="btn-item custom"
  onClick={[Function]}
  type="button"
>
  Test Button
</button>
`;
